### DISCUSSION TITLE
Add Peripheral Preferred Connection Parameter to Generic Access 0x1800

### DISCUSSION THREAD

**Discussion by mellbergsimon** [h2zero/NimBLE-Arduino#498] [Q&A] [answered]
Hello,

I was hoping someone had a deeper knowledge how to add a service to Generic Access 0x1800 Primary Service.

If i run:
```
#define SERVICE_UUID        "1800"

pService = pServer->createService(SERVICE_UUID");
```

Then when I view it inside BLE scanner it has created an additional service with the same name.

What I want to accomplish is changing the Peripheral Preferred Connection Parameter for the bluetooth module inside the created 0x1800 service. How does one accomplish this?

Kind Regards

**Comment by h2zero** [comment, marked as answer]
The creation of that service is handled by the stack already. To set the PPCP characteristic data you will need to define the values in the config somewhere. These are what you need to set:
```
#define MYNEWT_VAL_BLE_SVC_GAP_PPCP_MAX_CONN_INTERVAL (0)
#define MYNEWT_VAL_BLE_SVC_GAP_PPCP_MIN_CONN_INTERVAL (0)
#define MYNEWT_VAL_BLE_SVC_GAP_PPCP_SLAVE_LATENCY (0)
#define MYNEWT_VAL_BLE_SVC_GAP_PPCP_SUPERVISION_TMO (0)
```
